lib/rspec/core/shared_example_group.rb

```diff
@@ -180,6 +180,10 @@ def find(lookup_contexts, name)
           shared_example_groups[:main][name]
         end
 
+        def reset
+          shared_example_groups.delete_if { |k, _| k != :main }
+        end
```

Comment:
We would need a spec for this, as its important child shared example groups are still usable, its intentional they are not cleared (but unintentional that references to deleted example groups are kept around)